### ISSUE TITLE
API error handling 

### ISSUE THREAD

**Issue by jdogresorg** [jdogresorg/freewallet-desktop#26]
Need to update `cpRequest()` function to handle failures... currently only handling successes and no indication of error on failure.

Whenever a user performs an action, they should get some type of success or error message.

**Comment by jdogresorg** [comment]
Currently users are just seeing "Server Error" when the CP API throws an error, need to show the error message the API is returning

**Comment by jdogresorg** [comment]
better status updates in 6866ae4 .... closing

**Comment by jdogresorg** [comment]
reopening issue as users are still getting generalized errors at times.... need better error handling.

Here is an example error response which should help in adding better error handling

```
{
	"error": {
		"code": -32000,
		"data": {
			"type": "UnknownPubKeyError",
			"args": ["Public key was neither provided nor published in blockchain."]
		},
		"message": "Server error",
		"id": 0,
		"jsonrpc": "2.0"
	}
}
```

if error->data exists, display error `type` and `args` instead of `message`

**Comment by jdogresorg** [comment]
should be fixed with commit e8e99b1
